java-client/build.gradle.kts

```diff
@@ -337,7 +338,9 @@ publishing {
     }
 }
 
-if (JavaVersion.current() >= JavaVersion.VERSION_11) {
+// Use `-Pcheck-jdk8-compatibility=true` to
+val jdk8compatibility = (project.findProperty("check-jdk8-compatibility") as String?).toBoolean()
+if (JavaVersion.current() >= JavaVersion.VERSION_11 && jdk8compatibility == false) {
```

Comment:
I dislike the whole `check-jdk8-compatibility=true` business. Why do we need this at all and not just

```java
if (JavaVersion.current() >= JavaVersion.VERSION_11) {
 ...
} else if (JavaVersion.current() >= JavaVersion.VERSION_8) {
 ...
}
```

Reply:
That the thing, the `JavaVersion.current()` is the version Gradle run with, it is always 11 or higher. I will try to experiment with toolchains here.

Reply:
Ok, I see. 

Maybe we could improve on this by introducing optional `JAVA_TOOLCHAIN_VERSION` and default it to 11. Basically avoiding this whole `if` logic?

Stepping back, this does seem to work, so we should merge it and improve it later if you can't come up with something a lot nicer. Leave some TODOs with an explanation at the least?

Reply:
This is a good idea, thanks @dblock , I refactored it to use similar to OpenSearch core approach with `runtime.java` version.